contractsSol5/KyberFeeHandler.sol

```diff
@@ -361,4 +456,31 @@ contract KyberFeeHandler is IKyberFeeHandler, Utils4 {
         rebateWei = RRAmountWei * rebateInBps / BPS;
         rewardWei = RRAmountWei * rewardInBps / BPS;
     }
+
+    function validateEthToKncRateToBurn(uint rateEthToKnc) internal view returns(bool) {
+        require(rateEthToKnc <= MAX_RATE, "ethToKnc rate out of bounds");
+        require(rateEthToKnc > 0, "ethToKnc rate is 0");
+        require(sanityRateContract.length > 0, "no sanity rate contract");
+
+        // get latest knc/eth rate from sanity contract
+        uint kncToEthRate = sanityRateContract[0].latestAnswer();
+        require(kncToEthRate > 0, "sanity rate is 0");
+        require(kncToEthRate <= MAX_RATE, "sanity rate out of bounds");
+
+        uint sanityEthToKncRate = PRECISION * PRECISION / kncToEthRate;
+
+        // rate shouldn't be 10% lower than sanity rate
+        require(rateEthToKnc * BPS >= sanityEthToKncRate * (BPS - SANITY_RATE_DIFF_BPS), "Kyber Eth To KNC rate too low");
+
+        return true;
+    }
+
+    // check if an address is a contract or not
```

Comment:
Errrrr this can be hacked by doing calling the method in a constructor. 
See https://consensys.github.io/smart-contract-best-practices/recommendations/#avoid-using-extcodesize-to-check-for-externally-owned-accounts

Alternative is to use `tx.origin == msg.sender`, which I personally prefer.

Reply:
yeah the `tx.origin` check also has drawbacks :/ what do you think @ilanDoron ?

Reply:
seems for our case tx.origin check is better.

Reply:
I updated to use `tx.origin`